### PR TITLE
[Refactoring][log parser] ParserCompilerPass

### PR DESCRIPTION
**Pull request by ranpafin**
- Will use the container compiler functionality to add  all chain elements to JSONParser.
- Will make more explicit the existing order in the chain.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,5 @@ vendor
 composer.lock
 build/
 swiftmailer/
+.idea
+.idea/
```
Comment (Jean85) [inline .gitignore]: This changes will lower the SensioLabs score, can you please remove it?
You should use a global .gitIgnore for non project-related files like those; see https://help.github.com/articles/ignoring-files/

**Comment by Jean85** [comment]
Thanks! Nice refactoring!
Apart for my only comment, LGTM! :+1: Fix that and we're good to merge!

File: src/Paraunit/Parser/ParserCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     * @inheritdoc
+     */
+    public function process(ContainerBuilder $container)
+    {
+
+        $taggedServices = $container->findTaggedServiceIds(self::TAGGED_SERVICES);
+
+        uasort(
+            $taggedServices,
+            function ($a, $b) {
```
Comment (Jean85) [inline src/Paraunit/Parser/ParserCompilerPass.php]: Minor WS issue: https://scrutinizer-ci.com/g/facile-it/paraunit/inspections/e5c5a0ff-09c7-4330-9f85-dcf9dc969e67/patches?status=new

Comment (ranpafin) [inline src/Paraunit/Parser/ParserCompilerPass.php]: Let me fix this as well :+1:

**Comment by Jean85** [comment]
Thanks, that's great! Merging right now ;)
And I think I'll disable Coverall's comments, they're broken! :frowning: The check is enough.

**Comment by ranpafin** [comment]
@Jean85 yeah probably something related to the branch not being master...

**Comment by Jean85** [comment]
@ranpafin It may be related to https://github.com/satooshi/php-coveralls/issues/190

**Pull request merged**